pepper-apis/src/test/java/org/broadinstitute/ddp/route/GetUserAnnouncementsRouteTest.java

```diff
@@ -177,4 +189,74 @@ public void test_basicContentStyle() {
                 .body("$.size()", equalTo(1))
                 .body("[0].message", equalTo(msgPlainText));
     }
+
+    @Test
+    public void test_permanentAnnouncements_notDeletedAfterRead() {
+        TransactionWrapper.useTxn(handle -> handle.attach(UserAnnouncementDao.class)
+                .insert(testData.getUserId(), testData.getStudyId(), msgTemplate.getTemplateId(), true));
+
+        String guid = given().auth().oauth2(token)
+                .pathParam("userGuid", testData.getUserGuid())
+                .pathParam("studyGuid", testData.getStudyGuid())
+                .when().get(url)
+                .then().assertThat()
+                .statusCode(200).contentType(ContentType.JSON)
+                .body("$.size()", equalTo(1))
+                .body("[0].guid", not(isEmptyOrNullString()))
+                .body("[0].permanent", equalTo(true))
+                .body("[0].message", equalTo(msgTemplate.getTemplateText()))
+                .extract().path("[0].guid");
+
+        TransactionWrapper.useTxn(handle -> {
+            List<UserAnnouncement> found = handle.attach(UserAnnouncementDao.class)
+                    .findAllForUserAndStudy(testData.getUserId(), testData.getStudyId())
+                    .collect(Collectors.toList());
+            assertEquals(1, found.size());
+            assertEquals(guid, found.get(0).getGuid());
+        });
+
+        given().auth().oauth2(token)
+                .pathParam("userGuid", testData.getUserGuid())
+                .pathParam("studyGuid", testData.getStudyGuid())
+                .when().get(url)
+                .then().assertThat()
+                .statusCode(200).contentType(ContentType.JSON)
```

Comment:
There's a fair amount of copy-pasta here--can you factor this out to a private method?